Set proper java.specification.version

JasonFengJ9 opened the pull request:
Set proper `java.specification.version`

Removed the setting of `java.specification.version` by Java code;
Set `java.specification.version` to same value as `java.vm.specification.version` in C code.
Minor refactoring around code affected by this PR.

Now `Java 10 ./java -XshowSettings:properties -version 2>&1 | grep specification` output following:
```
    java.specification.name = Java Platform API Specification
    java.specification.vendor = Oracle Corporation
    java.specification.version = 10
    java.vm.specification.name = Java Virtual Machine Specification
    java.vm.specification.vendor = Oracle Corporation
    java.vm.specification.version = 10
``` 

closes: #2036 

Reviewer: @pshipton 
FYI: @DanHeidinga 

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

pshipton commented:
There is an obsolete vmi_getSystemProperty in vmifunc.c which has a comment about system properties that must be defined by the VM. We should add java.specification.version to the list.

pshipton commented:
Jason fyi https://github.com/eclipse/openj9/pull/2043#issuecomment-393684635 in case you missed it.

JasonFengJ9 commented:
Pete, thanks for the reminder, I was thinking to remove the obsolete methods which contains `IBM` words though was interrupted by another workitem periodically.

JasonFengJ9 commented:
Added the system property `java.specification.version` to the list, removed `IBM` from the comment, and updated other system property samples with latest `OpenJ9 Java 8` build.
@pshipton please have another look.

pshipton commented on runtime/jcl/common/stdinit.c:
```diff
@@ -827,8 +827,7 @@ computeBootstrapClassPathAppend(J9JavaVM *vm)
   *
-  * @note Requires that the 'java.home' and 'com.ibm.oti.system.class.path' system properties 
-  *            have been set in the VMI.
+  * @note Requires that the 'java.home' system properties have been set in the VMI.
```
"properties have" -> "property has"

JasonFengJ9 replied:
Yeah, fixed.

pshipton commented:
jenkins compile plinux

pshipton commented:
Started too many builds by accident. Killed the 8 & 11  builds.
